components/Paginated/PaginatedSuppliers.tsx

```diff
@@ -0,0 +1,75 @@
+import { FC } from 'react'
+
+import { Provider } from '@/services/GraphQL/providers/types'
+
+import styles from './Paginated.module.scss'
+
+interface PaginatedSuppliersProps {
+  providers: Provider[]
+  ProvidersPerPage: number
+  totalProvidersLength: number
+  setCurrentPage: (value: number) => void
+  CurrentPage: number
+  indexOfFirstProvider: number
+  indexOfLastProvider: number
+}
+const PaginatedSuppliers: FC<PaginatedSuppliersProps> = ({
+  ProvidersPerPage,
+  totalProvidersLength,
+  setCurrentPage,
+  CurrentPage,
+  providers,
+  indexOfFirstProvider,
+  indexOfLastProvider,
+}) => {
+  const pageNumbers = []
+  for (let i = 1; i <= Math.ceil(totalProvidersLength / ProvidersPerPage); i++) {
+    pageNumbers.push(i)
+  }
```

Comment:
What about use Array?

```suggestion
  const pageNumbers = Array.from({length: Math.ceil(totalProvidersLength / ProvidersPerPage)}, (_, index) => index + 1);
```

Or similar to create the array